Opi5 edge kernel support

efectn opened the pull request:
# Description
This PR adds some patches from Collabora and upstream kernel branch and add initial support for opi5

Jira reference number [AR-9999]

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] OPi5 boots fine
- [x] SD Card, GMAC, USB, PCIe2, Sata work

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

igorpecovnik commented:
OK. Any objections to merge this as is?

Possiblke paths:
- merging to Rockchip64 edge as is
- merging but moving patches to subfolder, put those into series perhaps?

efectn commented:
> OK. Any objections to merge this as is?
> 
> Possible paths:
> 
> * merging to Rockchip64 edge as is
> * merging but moving patches to subfolder, put those into series perhaps?
> * something else

Since rk3588 edge is pretty experimental, i don't prefer merging it to rockchip64 nowly. Perhaps we can merge on Linux 6.6 or 6.7

Btw most patches are mainlined on 6.5-rc1, should i update kernel?

Tonymac32 commented:
> Btw most patches are mainlined on 6.5-rc1, should i update kernel?

Yes, I think that's the correct path.  We should do our experiments/debugging with the Collabora kernel since it is the most feature-complete and intended for mainlining.  More or less to avoid a ton of work constantly updating the edge kernel given the speed the rk3588 support will be changing

efectn commented:
Updated kernel to 6.5-rc1 and removed mainlined patches.

> > Btw most patches are mainlined on 6.5-rc1, should i update kernel?
> 
> Yes, I think that's the correct path. We should do our experiments/debugging with the Collabora kernel since it is the most feature-complete and intended for mainlining. More or less to avoid a ton of work constantly updating the edge kernel given the speed the rk3588 support will be changing

I prefer working on edge branch because of it's under our controls and it's easier to add/remove patch from edge. Yes it's easier to use collabora branch but they have a lot of force pushes and i don't feel safe using it.

EvilOlaf commented:
Gave it a quick test. Builds, boots, installs to NVMe.
https://paste.armbian.com/ikavajufek

rpardini reviewed (approved):
merge it!

katyo commented:
@EvilOlaf NVMe stopped working on 6.7-rc1 for me. Have you test it?

igorpecovnik commented:
> Have you test it?

Please keep conversation at one place:
https://forum.armbian.com/forum/206-orange-pi-5-5b-5-plus/
